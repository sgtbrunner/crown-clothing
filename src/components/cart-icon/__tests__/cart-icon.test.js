import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import CartIcon from '../cart-icon.component';

const mockStore = configureMockStore();
const store = mockStore({
  cart: {
    cartItems: [
      {
        id: 1,
        price: 10,
        name: 'Item 1',
        quantity: 2,
        imageUrl: '',
      },
      {
        id: 2,
        price: 20,
        name: 'Item 2',
        quantity: 3,
        imageUrl: '',
      },
    ],
  },
});

const FIGURE = 'figure';

const mountComponent = () =>
  render(
    <Provider store={store}>
      <CartIcon />
    </Provider>
  );

describe('CartIcon component', () => {
  it('should render component snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render the icon within the component', () => {
    mountComponent();

    const cartIcon = screen.getByRole(FIGURE);
    expect(cartIcon).toBeInTheDocument();
    expect(cartIcon).toHaveTextContent('shopping-bag.svg');
  });

  it('should render 5 (five) as the item count when the redux store has 5 items', () => {
    mountComponent();

    const cartIcon = screen.getByRole(FIGURE);
    expect(cartIcon).toHaveTextContent(5);
  });
});
