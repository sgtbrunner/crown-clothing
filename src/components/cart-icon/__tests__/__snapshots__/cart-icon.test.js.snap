// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component should render component snapshot 1`] = `
<div>
  <div
    class="sc-bdnxRM gwlqxO"
    role="figure"
  >
    <svg
      class="sc-dlnjwi bSAYMN"
    >
      shopping-bag.svg
    </svg>
    <span
      class="sc-gtsrHT kIzdTM"
    >
      5
    </span>
  </div>
</div>
`;
